src/V3AstNodes.cpp

```diff
@@ -738,49 +738,66 @@ class AstNodeDType::CTypeRecursed final {
-        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true);
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
+        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true, false);
         info.m_type = "VlAssocArray<std::string, " + sub.m_type + ">";
     } else if (const auto* const adtypep = VN_CAST(dtypep, DynArrayDType)) {
-        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true);
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
+        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true, false);
         info.m_type = "VlQueue<" + sub.m_type + ">";
     } else if (const auto* const adtypep = VN_CAST(dtypep, QueueDType)) {
-        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true);
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
+        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true, false);
         info.m_type = "VlQueue<" + sub.m_type;
         // + 1 below as VlQueue uses 0 to mean unlimited, 1 to mean size() max is 1
         if (adtypep->boundp()) info.m_type += ", " + cvtToStr(adtypep->boundConst() + 1);
         info.m_type += ">";
     } else if (const auto* const adtypep = VN_CAST(dtypep, SampleQueueDType)) {
-        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true);
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
+        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(true, false);
         info.m_type = "VlSampleQueue<" + sub.m_type + ">";
     } else if (const auto* const adtypep = VN_CAST(dtypep, ClassRefDType)) {
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
         info.m_type = "VlClassRef<" + EmitCBase::prefixNameProtect(adtypep) + ">";
     } else if (const auto* const adtypep = VN_CAST(dtypep, IfaceRefDType)) {
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
         info.m_type = EmitCBase::prefixNameProtect(adtypep->ifaceViaCellp()) + "*";
     } else if (const auto* const adtypep = VN_CAST(dtypep, UnpackArrayDType)) {
+        UASSERT_OBJ(!packed, this, "Unsupported type for packed struct or union");
         if (adtypep->isCompound()) compound = true;
-        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(compound);
+        const CTypeRecursed sub = adtypep->subDTypep()->cTypeRecurse(compound, false);
         info.m_type = "VlUnpacked<" + sub.m_type;
         info.m_type += ", " + cvtToStr(adtypep->declRange().elements());
         info.m_type += ">";
-    } else if (VN_IS(dtypep, NodeUOrStructDType) && !VN_AS(dtypep, NodeUOrStructDType)->packed()) {
+    } else if (packed && (VN_IS(dtypep, PackArrayDType))) {
+        const auto* const adtypep = VN_CAST(dtypep, PackArrayDType);
```

Comment:
```suggestion
        const AstPackArrayDType* const adtypep = VN_CAST(dtypep, PackArrayDType);
```

We generally avoid autos for non-iterators to aid redability - similar below please.

Reply:
Fixed